processor/common/common.go

```diff
@@ -0,0 +1,340 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/spf13/cast"
+)
+
+var (
+	// TODO Add processor type invoking the BuildMatchProperties in error text.
+	errAtLeastOneMatchFieldNeeded = errors.New(
+		`error creating processor. At least one ` +
+			`of "services", "span_names" or "attributes" field must be specified"`)
+
+	errInvalidMatchType = fmt.Errorf(
+		`match_type must be either %q or %q`, MatchTypeStrict, MatchTypeRegexp)
+)
+
+// matchingProperties is an interface that allows matching a span against a configuration
+// of a match.
+type MatchingProperties interface {
+	MatchSpan(span *tracepb.Span, serviceName string) bool
+}
+
+type matchAttributes []matchAttribute
+
+// strictMatchingProperties allows matching a span against a "strict" match type
+// configuration.
+type strictMatchingProperties struct {
+	// Service names to compare to.
+	Services []string
+
+	// Span names to compare to.
+	SpanNames []string
+
+	// The attribute values are stored in the internal format.
+	Attributes matchAttributes
+}
+
+// strictMatchingProperties allows matching a span against a "regexp" match type
+// configuration.
+type regexpMatchingProperties struct {
+	// Precompiled service name regexp-es.
+	Services []*regexp.Regexp
+
+	// Precompiled span name regexp-es.
+	SpanNames []*regexp.Regexp
+
+	// The attribute values are stored in the internal format.
+	Attributes matchAttributes
+}
+
+// matchAttribute is a attribute key/value pair to match to.
+type matchAttribute struct {
+	Key            string
+	AttributeValue *tracepb.AttributeValue
+}
+
+func BuildMatchProperties(config *MatchProperties) (MatchingProperties, error) {
```

Comment:
```suggestion
func BuildSpanPropertyMatcher(config *MatchProperties) (SpanPropertyMatcher, error) {
```

Reply:
If we go with `spanmatcher` this can be become `New(config *Config) (Matcher, error)` just spelling it out so we can contrast the alternatives.

Reply:
The other suggestion is `span` so this becomes `NewMatcher(config *MatcherConfig) (Matcher, error)`

Reply:
NewMatcher is what I went with since it is under the span package now.